### PR TITLE
Some small cleanups in `rustc_abi`

### PR DESCRIPTION
**Pull request by nnethercote**
Minor things I found while looking at this crate's code.

r? @the8472

### CODE REVIEW

File: compiler/rustc_abi/src/layout.rs [hunk 1]
```diff
@@ -868,18 +869,14 @@ fn univariant(
...
         // If `-Z randomize-layout` was enabled for the type definition we can shuffle
         // the field ordering to try and catch some code making assumptions about layouts
-        // we don't guarantee
+        // we don't guarantee.
         if repr.can_randomize_type_layout() && cfg!(feature = "randomize") {
-            #[cfg(feature = "randomize")]
-            {
-                // `ReprOptions.layout_seed` is a deterministic seed that we can use to
-                // randomize field ordering with
-                let mut rng = Xoshiro128StarStar::seed_from_u64(repr.field_shuffle_seed.as_u64());
```
Comment (the8472) [inline compiler/rustc_abi/src/layout.rs:877]: The `Xoshiro128StarStar` import is conditional on that feature.

Comment (nnethercote) [reply compiler/rustc_abi/src/layout.rs]: Hmm, I see. I will remove that commit.

Now I'm wondering why the "randomize" feature exists. I bet it's never disabled within rustc itself. rust-analyzer, maybe?

**Comment by nnethercote** [comment]
I have updated the code, removing the commit touching the feature gates.

File: compiler/rustc_abi/src/layout.rs (hunk 1 above)
Comment (Veykril) [reply compiler/rustc_abi/src/layout.rs]: It is currently disabled in rust-analyzer yes, and now that it's depended on the `Hash64` type of `rustc_data_structures` r-a is unlikely to ever enable it as we don't want to pull in that crate.

**Pull request merged**